### DISCUSSION TITLE
Javascript in pdfjs

### DISCUSSION THREAD

**Discussion by mlmatlock** [qutebrowser/qutebrowser#6422] [General] [answered]
With Mozilla making the (IMO) bone-headed decision to enable javascript by default in the firefox native pdf viewer (pdfjs), should we assume that Mozilla is doing the same for the stand-alone pdfjs used by Qutebrowser?

If so, how would you turn that off - in firefox, you set ```pdfjs.enableScripting``` to false.

**Comment by The-Compiler** [comment, marked as answer]
Indeed the default has been changed to `true` inside PDF.js: https://github.com/mozilla/pdf.js/pull/13116

That's not possible to configure right now, I opened #6424. However, as long as you have JS enabled for websites, I don't really see a problem with allowing it for PDFs too, no?

**Reply by mlmatlock** [reply]
I'm not too sure; there's a pretty good debate on [slashdot](https://news.slashdot.org/story/21/04/22/1754258/firefox-88-enables-javascript-embedded-in-pdfs-by-default) about this.  Personally, I think that this is a bad idea (think macros in MS Office documents).

For example:  Now, instead of just reading a dry, crusty pdf on some obscure systemd function, now I have to be concerned if someone put some kind of script in it (that will run unconditionally), and don't even know if (if it even exists) pdfjs's sandboxing mechanism might work.
